contrib/delta-kernel-rs-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+set(DELTA_KERNEL_RS_SOURCE_DIR "${ClickHouse_SOURCE_DIR}/contrib/delta-kernel-rs")
+set(DELTA_KERNEL_RS_BINARY_DIR "${CMAKE_BINARY_DIR}/contrib/delta-kernel-rs")
+set(DELTA_KERNEL_RS_CMAKE_DIR "${ClickHouse_SOURCE_DIR}/contrib/delta-kernel-rs-cmake")
+
+include(ExternalProject)
+
+
+set(ENABLE_DELTA_KERNEL_RS TRUE)
+if (NOT ENABLE_LIBRARIES)
+  set(ENABLE_DELTA_KERNEL_RS FALSE)
+else()
+  list (APPEND CMAKE_MODULE_PATH "${ClickHouse_SOURCE_DIR}/contrib/delta-kernel-rs/cmake")
+  find_package(Rust)
+  set(DEFAULT_ENABLE_RUST ${Rust_FOUND})
+endif()
+
+option(ENABLE_DELTA_KERNEL_RS "Enable delta-kernel-rs" ${DELTA_KERNEL_RS})
+if (NOT ENABLE_DELTA_KERNEL_RS)
+  message(STATUS "Not using delta-kernel-rs")
+  return()
+endif()
+
+set(EXTENSION_SOURCES ${DELTA_KERNEL_RS_SOURCE_DIR}/delta_extension.cpp)
+
+add_library(_delta_kernel_rs STATIC ${EXTENSION_SOURCES})
+
+set(DELTA_KERNEL_TARGET_PATH ${DELTA_KERNEL_RS_BINARY_DIR}/target4)
+
+ExternalProject_Add(
```

Comment:
Can't we add this sub-module to `./rust` folder and build it along with other rust libraries we use? Or does its build process differ significantly?